lib/SPIRV/SPIRVInternal.h

```diff
@@ -1098,6 +1098,97 @@ bool postProcessBuiltinsWithArrayArguments(Module *M, bool IsCpp = false);
 
 template <typename T>
 MetadataAsValue *map2MDString(LLVMContext &C, SPIRVValue *V);
+
+namespace detail {
+template <typename T, std::size_t SizeOfT> struct LeadingZerosCounter {
+  static unsigned count(T Val) {
+    if (!Val)
+      return std::numeric_limits<T>::digits;
+
+    // Bisection method.
+    unsigned ZeroBits = 0;
+    for (T Shift = std::numeric_limits<T>::digits >> 1; Shift; Shift >>= 1) {
+      T Tmp = Val >> Shift;
+      if (Tmp)
+        Val = Tmp;
+      else
+        ZeroBits |= Shift;
+    }
+    return ZeroBits;
+  }
```

Comment:
shouldn't it be just the entire algorithm (with few additions, like if <2 return 1 and if already power of 2 return Val?):
```suggestion
    Val--;
    for (T Shift = std::numeric_limits<T>::digits >> 1; Shift; Shift >>= 1) {
      Val | = Val >> Shift;
    }
    return Val++;
  }
```